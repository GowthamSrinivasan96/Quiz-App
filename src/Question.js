import React, { useState, useEffect } from 'react';
import './App.scss';
import './bootstrap.min.css';

function Question(props) {
return (
    <>
    <form class="form-signin" autocomplete="off" key={props.quizId}>
        {props.timer > 0 && props.lastPage !== 'yes'? 
            <> 
            <div className="form-group question-header">
                {props.index>=0 && props.qno.length && props.qno[props.index] ? props.qno[props.index].question:''}
            </div>
            <div className="flex-container">
                {props.qno.length && props.index>=0 && props.index < props.qno.length ?(props.qno[props.index].options.map((option)=>
                    <div key={option} id={option} className="col-md-5"  onClick={() =>{props.onclick(props.qno[props.index],option)}}>
                        {option}
                    </div>
                )):''} 
            </div>  
            </> : ''}
     </form>
     <div class="login-footer">
        <div class="banner-text-container">
            <div class="login-copyright-notice">{props.crctAns === 'yes' ?'Correct Answer':(props.crctAns === 'no'? 'Wrong Answer':'')}</div>
        </div>
     </div> 
    </>                    
)
}

export default Question;