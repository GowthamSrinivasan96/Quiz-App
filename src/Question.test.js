import React from 'react';
import { render } from '@testing-library/react';
import Question from './Question';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';


it('matches the snapshot', () => {
  const tree = renderer.create(<Question />).toJSON();
  expect(tree).toMatchSnapshot();
});