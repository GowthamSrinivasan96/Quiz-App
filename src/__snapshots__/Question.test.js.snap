// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
Array [
  <form
    autocomplete="off"
    class="form-signin"
  >
    
  </form>,
  <div
    class="login-footer"
  >
    <div
      class="banner-text-container"
    >
      <div
        class="login-copyright-notice"
      >
        
      </div>
    </div>
  </div>,
]
`;
